src/Security.php

```diff
@@ -24,296 +26,50 @@
-        $str = preg_replace_callback(
-            '/<\w+.*?(?=>|<|$)/si',
-            Utils::decodeEntityCallback($hash),
-            $str
-        );
-
-        $str = self::removeInvisibleCharacters($str);
-
-        $str = str_replace("\t", ' ', $str);
-
-        $str = Utils::doNeverAllowed($str);
-
-        $str = str_replace(['<?', '?'.'>'], ['&lt;?', '?&gt;'], $str);
-
-        $words = [
-            'javascript', 'expression', 'vbscript', 'jscript', 'wscript',
-            'vbs', 'script', 'base64', 'applet', 'alert', 'document',
-            'write', 'cookie', 'window', 'confirm', 'prompt', 'eval',
-        ];
-
-        foreach ($words as $word) {
-            $word = implode('\s*', str_split($word)).'\s*';
-            $str = preg_replace_callback(
-                '#('.substr($word, 0, -3).')(\W)#is',
-                [$this, 'compactExplodedWords'],
-                $str
-            );
-        }
-
-        do {
-            $original = $str;
-
-            if (preg_match('/<a/i', $str)) {
-                $str = preg_replace_callback(
-                    '#<a[^a-z0-9>]+([^>]*?)(?:>|$)#si',
-                    [$this, 'jsLinkRemoval'],
-                    $str
-                );
-            }
-
-            if (preg_match('/<img/i', $str)) {
-                $str = preg_replace_callback(
-                    '#<a(?:rea)?[^a-z0-9>]+([^>]*?)(?:>|$)#si',
-                    [$this, 'jsImgRemoval'],
-                    $str
-                );
-            }
-
-            if (preg_match('/script|xss/i', $str)) {
-                $str = preg_replace('#</*(?:script|xss).*?>#si', '[removed]', $str);
-            }
-        } while ($original !== $str);
-
-        unset($original);
-
-        $pattern = '#'
-            .'<((?<slash>/*\s*)((?<tagName>[a-z0-9]+)(?=[^a-z0-9]|$)|.+)'
-            .'[^\s\042\047a-z0-9>/=]*'
-            .'(?<attributes>(?:[\s\042\047/=]*'
-            .'[^\s\042\047>/=]+'
-                .'(?:\s*='
-                    .'(?:[^\s\042\047=><`]+|\s*\042[^\042]*\042|\s*\047[^\047]*\047|\s*(?U:[^\s\042\047=><`]*))'
-                .')?'
-            .')*)'
-            .'[^>]*)(?<closeTag>\>)?#isS';
-
-        do {
-            $original = $str;
-            $str = preg_replace_callback($pattern, Utils::sanitizeHtmlCallback($this->evil), $str);
-        } while ($original !== $str);
-
-        unset($original);
-
-        $str = preg_replace(
-            '#(alert|prompt|confirm|cmd|passthru|eval|exec|expression|system|fopen|fsockopen|file|file_get_contents|readfile|unlink)(\s*)\((.*?)\)#si',
-            '\\1\\2&#40;\\3&#41;',
-            $str
-        );
-
-        $str = preg_replace(
-            '#(alert|prompt|confirm|cmd|passthru|eval|exec|expression|system|fopen|fsockopen|file|file_get_contents|readfile|unlink)(\s*)`(.*?)`#si',
-            '\\1\\2&#96;\\3&#96;',
-            $str
-        );
-
-        return Utils::doNeverAllowed($str);
+        $this->antiXss->setReplacement($replacement);
```

Comment:
Hmmm, I kinda don't like setting stuff on an object that was passed to the constructor too. Seems a bit odd.

Reply:
moved in separate method